src/main/java/seedu/address/commons/util/CsvUtil.java

```diff
@@ -36,39 +42,121 @@ public class CsvUtil {
     public static final String MESSAGE_ERROR_READING_FILE = "Error reading file: ";
 
     /**
-     * Reads the CSV file and returns a list of maps,
-     * where each map represents a row of person's data in the csv file.
-     * @throws DataLoadingException
+     * Reads a CSV file.
+     * Firstly, it checks if the compulsory parameters are present in the header of the csv file (the header
+     * is the first line of the csv file).
+     * Secondly, it only reads the columns that are in compulsoryParameters or optionalParameters.
+     * Values in other headers will be ignored.
+     * Thirdly, it checks if the number of values in each row is the same as the number of headers.
+     * If the check for compulsory parameters fails, the error report will contain the missing compulsory parameters.
+     * If the check for the number of values in each row fails, the error report will contain the rows that do not have
+     * the same number of values as the number of headers.
+     * It is crucial that a dataLoadingException is thrown in the
+     * caller of this method if the error report is not empty.
+     * @param filePath the path of the file
+     * @param compulsoryParameters
+     * @param optionalParameters
+     * @return
+     *     A pair in which the first value is the data and the second is the error report. Data can be null if
+     *     the file is not read successfully or if any compulsory parameters are missing in the header row of csv file.
+     * @throws IOException
      */
-    public static List<Map<String, String>> readCsvFile(Path filePath) throws DataLoadingException {
+    public static Pair<Optional<List<Map<String, String>>>, String> readCsvFile(
+            Path filePath, HashSet<String> compulsoryParameters, HashSet<String> optionalParameters)
+            throws IOException {
+        List<Map<String, String>> data = null;
+        Optional<List<Map<String, String>>> nullableData = Optional.empty();
+        String errorMsgs = "";
         try {
             CSVReader reader = new CSVReaderBuilder(new FileReader(filePath.toString())).build();
+            List<String> headers = List.of(reader.readNext()); // first line should be headers
+            List<Integer> columnsToSkip = coloumnsToSkip(headers, compulsoryParameters, optionalParameters);
             List<String[]> rows = reader.readAll();
-            return parseData(rows);
-        } catch (IOException | CsvException e) {
-            throw new DataLoadingException(e);
+            Pair<List<Map<String, String>>, String> result = parseData(rows, headers, columnsToSkip);
+            data = result.getKey();
+            errorMsgs = result.getValue();
+            nullableData = Optional.of(data);
+            return new Pair<>(nullableData, errorMsgs);
+        } catch (DataLoadingException | CsvException e) {
+            return new Pair<>(nullableData, e.getMessage());
```

Comment:
The return type of this function is a little complicated. As mentioned in a previous comment, it might be a good idea to abstract this (Including the fact that data might be empty) into a seperate class.